### ISSUE TITLE
Unable to delete failed deployment

### ISSUE THREAD

**Issue by SpartyInDetroit** [Azure/AKS#2121]
**What happened**:

`az aks delete` is unable to delete my AKS cluster. This also fails when attempting to delete from the portal. Attempting to delete the resource group also fails but with a different error.

**What you expected to happen**:

AKS cluster should be destroyed.

**How to reproduce it (as minimally and precisely as possible)**:

I believe it's in a strange state because it was a private cluster and the managed identity and private dns zone that were created before it were deleted first.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.18.4
- Size of cluster (how many worker nodes are in the cluster?): 3 failed nodes
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.)
- Others: Fresh deployment. Was troubleshooting creating a private cluster on a vnet with byo dns.

**Comment by aravindarc** [comment]
@SpartyInDetroit I am facing the same issue, I created the cluster through terraform, the private dns got deleted first. But when I try to delete the cluster from portal, the following error comes up.

`Error: privateDNSVNetLinkReconciler retry failed: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to...`

Can you confirm if this is the same issue you are facing. And did you find a solution by any chance?

**Comment by SpartyInDetroit** [comment]
@aravindarc yes! This is the exact same issue I am getting.

**Comment by IanMoroney** [comment]
You could try updating/creating a new sp for the cluster:
https://docs.microsoft.com/en-us/azure/aks/update-credentials
and once complete, try to delete the cluster again?
The issue is most likely an ordering problem, where the SP gets deleted before it has a chance to delete the MC_ resources.

**Comment by aravindarc** [comment]
Yes, let me tell you why this happens. So when you try bringing up a private cluster with terraform. when there is any issue with aks creation, it fails and the aks is not registered in the terraform state. So when you try destroying the terraform resources. the private dns zone registered with the aks gets destroyed but the aks is left as it is. After that when you try to delete it from portal it will not work because it cannot resolve the api server with private dns. So to avoid this, if you find that aks is not registered in the terraform state, before doing terraform destroy, delete the aks cluster from portal followed by terraform destroy.

Now to delete the hanging aks cluster, you cannot do that. You have to raise a support ticket to MS, they'll help you delete the resource.

**Comment by miwithro** [comment]
@SpartyInDetroit are you still having the issue?  Were you able to open a ticket with support?
